### PR TITLE
feat(b-modal): for accessibility, read only modal title and not whole header + additional A11Y options (addresses #3712)

### PR DESCRIPTION
**Pull request by tmorehouse**
### Describe the PR

Addresses https://github.com/bootstrap-vue/bootstrap-vue/issues/3712#issuecomment-513049766

Sets `aria-labelledby` to point to the modal header title, instead of the full modal header, which prevents the `close` button being announced by screen readers as part of the label for the modal.

Also adds in a new prop `title-sr-only` to visually hide the header title but still make it available to screen readers (applicable only when `hide-header` is not set), and adds new prop `aria-label` (string) which will be used instead of the modal's title (handy if `hide-header` is set).

Adds new prop `title-class` for adding class(es) to the modal's title element.

To Do:

- [x] add prop `aria-label` to use as an alternative _instead_ of the modal's title
- [x] documentation update (mainly accessibility section)

### PR checklist

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [x] Enhancement
- [x] ARIA accessibility
- [ ] Documentation update
- [ ] Other (please describe)

**Does this PR introduce a breaking change?** (check one)

- [x] No
- [ ] Yes (please describe)

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch, **not** the `master` branch
- [x] When resolving a specific issue, it's referenced in the PR's title (i.e. `[...] (fixes #xxx[,#xxx])`, where "xxx" is the issue number)
- [x] It should address only one issue or feature. If adding multiple features or fixing a bug and adding a new feature, break them into separate PRs if at all possible.
- [x] The title should follow the [**Conventional Commits**](https://www.conventionalcommits.org/) naming convention (i.e. `fix(alert): not alerting during SSR render`, `docs(badge): update pill examples, fix typos`, `chore: fix typo in README`, etc). **This is very important, as the `CHANGELOG` is generated from these messages.**

**If new features/enhancement/fixes are added or changed:**

- [x] Includes documentation updates (including updating the component's `package.json` for slot and event changes)
- [ ] New/updated tests are included and passing (if required)
- [x] Existing test suites are passing
- [x] The changes have not impacted the functionality of other components or directives
- [x] ARIA Accessibility has been taken into consideration (Does it affect screen reader users or keyboard only users? Clickable items should be in the tab index, etc.)

**If adding a new feature, or changing the functionality of an existing feature, the PR's
description above includes:**

- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

### CODE REVIEW

File: src/components/modal/modal.js [hunk 1]
```diff
@@ -720,9 +720,15 @@ export const BModal = /*#__PURE__*/ Vue.extend({
               ? { innerHTML: this.titleHtml }
               : {}
           modalHeader = [
-            h(this.titleTag, { class: ['modal-title'], domProps }, [
-              this.normalizeSlot('modal-title', this.slotScope) || stripTags(this.title)
-            ]),
+            h(
+              this.titleTag,
+              {
+                class: ['modal-title'],
```
Comment (jackmu95) [inline src/components/modal/modal.js]: staticClass : 'modal-title',

**Comment by tmorehouse** [comment]
Thinking of adding a new prop `modal-title-sr-only` to visually hide the modal title, which would still make a label available to screen reader only users

**Comment by tmorehouse** [comment]
Added the prop `title-sr-only` for visually hiding the title, but still making it accessible to screen readers.

**Comment by tmorehouse** [comment]
Will be adding in a prop `aria-label` as an alternative to using the modal header content as the modal label for screen readers.

File: src/components/modal/modal.js (hunk 1 above)
Comment (tmorehouse) [reply src/components/modal/modal.js]: Fixed

**Pull request merged**